kotlin/vci-client/src/main/java/io/mosip/vciclient/authorizationCodeFlow/interactiveAuthorization/handler/InteractiveAuthorizationHandler.kt

```
@@ -0,0 +1,174 @@
+        traceabilityId: String? = null
+    ): InteractionResponse {
+
+        val parsedPresentationInteractionResponse = JsonUtils.deserialize(
+            presentationInteractionResponse,
+            PresentationInteractionResponse::class.java
+        )
+            ?: throw InteractiveAuthorizationException("Failed to parse presentation interaction response")
+
+        try {
+            parsedPresentationInteractionResponse.validate()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)